### PR TITLE
One shot sign/verify DSA

### PR DESCRIPTION
**Pull request by palaviv**
I just did the same as in #2945 for DSA.

### CODE REVIEW

**Comment by alex** [comment]
Jenkins, OK to test

File: docs/hazmat/primitives/asymmetric/dsa.rst [hunk 1]
```diff
@@ -289,6 +309,21 @@ Key interfaces
 
         The bit length of :attr:`~DSAParameterNumbers.q`.
 
+    .. method:: sign(data, algorithm)
+
+        .. versionadded:: 1.4
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/dsa.rst]: This needs to be 1.5

File: docs/hazmat/primitives/asymmetric/dsa.rst [hunk 2]
```diff
@@ -400,6 +435,24 @@ Key interfaces
 
         :return bytes: Serialized key.
 
+    .. method:: verify(signature, data, algorithm)
+
+        .. versionadded:: 1.4
```
Comment (reaperhulk) [inline docs/hazmat/primitives/asymmetric/dsa.rst]: This needs to be 1.5 as well

**Comment by reaperhulk** [comment]
Could you also add a changelog entry (in the style of what was done for the new RSA methods in 1.4) for this? You'll probably need to rebase to avoid a conflict there.

File: docs/hazmat/primitives/asymmetric/dsa.rst (hunk 1 above)
Comment (palaviv) [inline docs/hazmat/primitives/asymmetric/dsa.rst]: done

File: docs/hazmat/primitives/asymmetric/dsa.rst (hunk 2 above)
Comment (palaviv) [inline docs/hazmat/primitives/asymmetric/dsa.rst]: done

**Comment by palaviv** [comment]
Added the changelog entry.

**Pull request merged**